lib/rubygems.rb

```diff
@@ -941,7 +941,7 @@ def self.ui
   def self.use_paths(home, *paths)
     paths = nil if paths == [nil]
     paths = paths.first if Array === Array(paths).first
-    self.paths = { "GEM_HOME" => home, "GEM_PATH" => paths }
+    self.paths = { "GEM_HOME" => home, "GEM_PATH" => paths.join(Gem.path_separator) }
```

Comment:
Do we know why it's a regexp on JRuby? That seems strange, but I'll fix
this in the mean time. /cc @headius
On Sat, Feb 6, 2016 at 12:03 PM Samuel E. Giddins <notifications@github.com>
wrote:

> In lib/rubygems.rb
> <https://github.com/rubygems/rubygems/pull/1094#discussion_r52106703>:
>
> > @@ -941,7 +941,7 @@ def self.ui
> >    def self.use_paths(home, *paths)
> >      paths = nil if paths == [nil]
> >      paths = paths.first if Array === Array(paths).first
> > -    self.paths = { "GEM_HOME" => home, "GEM_PATH" => paths }
> > +    self.paths = { "GEM_HOME" => home, "GEM_PATH" => paths.join(Gem.path_separator) }
>
> this when fail when Gem.path_separator is a regex, as it is on Jruby
> @tenderlove <https://github.com/tenderlove> (see #1469
> <https://github.com/rubygems/rubygems/pull/1469>)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rubygems/rubygems/pull/1094/files#r52106703>.
>